### DISCUSSION TITLE
✅ how does webssh keep connections alive?

### DISCUSSION THREAD

**Discussion by jeff-geoff** [isontheline/pro.webssh.net#428] [Q&A] [answered]
Other SSH apps terminate connections when they are switched to the background. How does webssh prevent iOS from pausing it and closing the connection?

**Comment by isontheline** [comment, marked as answer]
As all other apps, iOS kills WebSSH connections after 30 seconds in background.

One exception : tunneling (using VPN-Over-SSH)

**Comment by jeff-geoff** [comment]
I have not found that to be the case on iOS 15

my connections persist for far longer than the typical 30 seconds of other apps.

**Reply by isontheline** [reply]
Perhaps a new behavior (or bug) from Apple.
I didn't heard about this yet.
Does an app like Termius have connections killed between the 30 seconds allowed?

**Reply by jeff-geoff** [reply]
Terminus kills connections after 30 seconds as before.

I only started using webssh recently, so I have never seen it behave differently.

I’ve had a webssh connection open since you posted your first reply and it is still active as of now.

**Reply by isontheline** [reply]
Strange 🧐

Perhaps Termius team doesn't have recompiled their app with the latest Xcode 13.

Nevertheless this (new) behavior wasn't introduced by myself though several users could like this 😇

**Reply by jeff-geoff** [reply]
I am one of those users. I hope it is not a bug!
